### PR TITLE
dist/redhat: drop rpm conflict with ABRT, add systemd conflict instead

### LINKED ISSUE #892: Remove ARBT conflict
Instead of conflict Scylla rpm with arbt, do the following:

BlackListedPaths = /usr/share/doc/_, */example_
Crashes in executables in these paths will be ignored by ABRT.

### PR DESCRIPTION
**Pull request by syuu1228**
Currently, "yum install scylla" causes conflict when ABRT is installed.

To avoid this behavior and keep using systemd-coredump for scylla coredump, let's drop "Conflicts: abrt" from rpm and add "Conflicts=abrt-ccpp.service" to systemd unit.

Fixes #892

### CODE REVIEW

File: dist/common/scripts/scylla_coredump_setup
Comment (tchaikov) [inline dist/common/scripts/scylla_coredump_setup:92]: instead rewriting the saved file with the existing settings to make `/usr/sbin/abrt-install-ccpp-hook uninstall` a no-op. probably, a simpler solution is to just remove `/var/run/abrt/saved_core_pattern`.

File: dist/common/systemd/scylla-server.service
Comment (tchaikov) [inline dist/common/systemd/scylla-server.service:6]: abrt-ccpp.service runs abrt-hook-ccpp. the former was removed from abrt in v 2.14.0. see https://github.com/abrt/abrt/blob/90c3957bd644ded992a36672069cdcbe0f60b3f5/abrt.spec#L1320 and the latter was remove and the latter was remove in v 2.13.0, see https://github.com/abrt/abrt/blob/90c3957bd644ded992a36672069cdcbe0f60b3f5/CHANGELOG.md?plain=1#L154 .the `abrt-hook-ccpp` script overwrites the core dump pattern file so the coredump will be processed by its own `abrt-hook-ccpp` . the abrt upstream ditched the hack in favor of systemd-coredump.

since we still need to support el7 and el8, which come with abrt-2.10, we cannot drop this change yet. but might want to note this as a comment in code, that this only applies to abrt < 2.14 .

File: dist/common/scripts/scylla_coredump_setup
Comment (tchaikov) [reply dist/common/scripts/scylla_coredump_setup:88]: or, in addition to conflict with `abrt-ccpp.service`, we can just stop the service unit in this script **before**  setting up the pattern file and update the system with it, so that the `abrt-ccpp.service` never rewrites our settings. simpler this way.

Comment (syuu1228) [reply dist/common/scripts/scylla_coredump_setup]: f-string was typo, dropped.
The fix was removing strange " after %e:
`'kernel.core_pattern=|/usr/lib/systemd/systemd-coredump %p %u %g %s %t %e"'`

File: dist/common/scripts/scylla_coredump_setup
Comment (syuu1228) [reply dist/common/scripts/scylla_coredump_setup]: Dropped previous code, rerwrite it to stopping/disabling abrt-ccpp.service.

File: dist/common/systemd/scylla-server.service
Comment (syuu1228) [reply dist/common/systemd/scylla-server.service:7]: Added comment "This will only requires for abrt < 2.14"

File: dist/common/systemd/scylla-server.service
Comment (tchaikov) [inline dist/common/systemd/scylla-server.service:6]: is required.

**Review by tchaikov — APPROVED** [review]
lgtm modulo a nit

**Comment by tchaikov** [comment]
weird. the tests are failing due to core dumps.

**Comment by tchaikov** [comment]
[test_snapshot_options_matrix](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4530/testReport/junit/test.nodetool/test_snapshot/Tests___Unit_Tests___test_snapshot_options_matrix_False_ktlist5_None_) this failure was caused by #15939

**Comment by tchaikov** [comment]
@scylladb/scylla-maint hi maintainers, could you help queue this change?

@avikivity do you want to take a final look?

**Pull request closed without merging**
